src/network/netdev/netdev.c

```diff
@@ -401,28 +401,24 @@ int netdev_set_ifindex(NetDev *netdev, sd_netlink_message *message) {
         if (r < 0)
                 return log_netdev_error_errno(netdev, r, "Could not get rtnl message type: %m");
 
-        if (type != RTM_NEWLINK) {
-                log_netdev_error(netdev, "Cannot set ifindex from unexpected rtnl message type.");
-                return -EINVAL;
-        }
+        if (type != RTM_NEWLINK)
+                return log_netdev_error_errno(netdev, SYNTHETIC_ERRNO(EINVAL), "Cannot set ifindex from unexpected rtnl message type.");
 
         r = sd_rtnl_message_link_get_ifindex(message, &ifindex);
         if (r < 0) {
-                log_netdev_error_errno(netdev, r, "Could not get ifindex: %m");
                 netdev_enter_failed(netdev);
-                return r;
+                return log_netdev_error_errno(netdev, r, "Could not get ifindex: %m");
```

Comment:
I think we should keep the original sequence of operations here. netdev_enter_failed() does a bunch of work, in particular this results in link_unref(), which might or might not trigger a cascade on unrefs. Even if that code doesn't result in anything user-visible right now, it might start doing so in the future. It seems better to immediately log about the issue, and only start doing this cleanup work later.

Reply:
OK. Dropped.